<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>983</width>
    <height>577</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>MainWindow</string>
  </property>
  <widget class="QWidget" name="centralwidget">
   <layout class="QVBoxLayout" name="verticalLayout_6">
    <item>
     <widget class="QSplitter" name="splitter_3">
      <property name="orientation">
       <enum>Qt::Vertical</enum>
      </property>
      <widget class="QSplitter" name="splitter">
       <property name="orientation">
        <enum>Qt::Horizontal</enum>
       </property>
       <widget class="CubeWidget" name="cubewidget" native="true">
        <property name="minimumSize">
         <size>
          <width>640</width>
          <height>480</height>
         </size>
        </property>
        <property name="focusPolicy">
         <enum>Qt::StrongFocus</enum>
        </property>
       </widget>
       <widget class="QWidget" name="layoutWidget">
        <layout class="QVBoxLayout" name="verticalLayout_5">
         <item>
          <layout class="QVBoxLayout" name="verticalLayout">
           <item>
            <layout class="QHBoxLayout" name="horizontalLayout_2">
             <item>
              <widget class="QLabel" name="fps_label">
               <property name="font">
                <font>
                 <family>Malgun Gothic</family>
                 <weight>50</weight>
                 <bold>false</bold>
                </font>
               </property>
               <property name="text">
                <string>FPS:</string>
               </property>
               <property name="textFormat">
                <enum>Qt::PlainText</enum>
               </property>
               <property name="scaledContents">
                <bool>true</bool>
               </property>
               <property name="margin">
                <number>0</number>
               </property>
               <property name="openExternalLinks">
                <bool>false</bool>
               </property>
              </widget>
             </item>
             <item>
              <widget class="QLabel" name="fps">
               <property name="text">
                <string>NAN</string>
               </property>
              </widget>
             </item>
            </layout>
           </item>
           <item>
            <widget class="QLabel" name="direct_label">
             <property name="font">
              <font>
               <weight>75</weight>
               <bold>true</bold>
              </font>
             </property>
             <property name="text">
              <string>Direct Light</string>
             </property>
            </widget>
           </item>
           <item>
            <widget class="QCheckBox" name="direct_state">
             <property name="text">
              <string>On / Off</string>
             </property>
            </widget>
           </item>
           <item>
            <layout class="QHBoxLayout" name="horizontalLayout_6">
             <item>
              <widget class="QLabel" name="direct_intensity_label">
               <property name="text">
                <string>I</string>
               </property>
              </widget>
             </item>
             <item>
              <widget class="QSlider" name="direct_intensity">
               <property name="layoutDirection">
                <enum>Qt::RightToLeft</enum>
               </property>
               <property name="orientation">
                <enum>Qt::Horizontal</enum>
               </property>
              </widget>
             </item>
            </layout>
           </item>
           <item>
            <layout class="QHBoxLayout" name="horizontalLayout">
             <item>
              <widget class="QLabel" name="direct_x">
               <property name="text">
                <string>x</string>
               </property>
              </widget>
             </item>
             <item>
              <widget class="QSlider" name="direct_x_slider">
               <property name="orientation">
                <enum>Qt::Horizontal</enum>
               </property>
              </widget>
             </item>
             <item>
              <widget class="QLabel" name="direct_y">
               <property name="text">
                <string>y</string>
               </property>
              </widget>
             </item>
             <item>
              <widget class="QSlider" name="direct_y_slider">
               <property name="orientation">
                <enum>Qt::Horizontal</enum>
               </property>
              </widget>
             </item>
             <item>
              <widget class="QLabel" name="direct_z">
               <property name="text">
                <string>z</string>
               </property>
              </widget>
             </item>
             <item>
              <widget class="QSlider" name="direct_z_slider">
               <property name="orientation">
                <enum>Qt::Horizontal</enum>
               </property>
              </widget>
             </item>
            </layout>
           </item>
          </layout>
         </item>
         <item>
          <layout class="QVBoxLayout" name="verticalLayout_2">
           <item>
            <widget class="QPushButton" name="direct_change_color_button">
             <property name="minimumSize">
              <size>
               <width>0</width>
               <height>28</height>
              </size>
             </property>
             <property name="text">
              <string>Change color</string>
             </property>
            </widget>
           </item>
           <item>
            <widget class="QLabel" name="point_label">
             <property name="font">
              <font>
               <weight>75</weight>
               <bold>true</bold>
              </font>
             </property>
             <property name="text">
              <string>Point Light</string>
             </property>
            </widget>
           </item>
           <item>
            <widget class="QCheckBox" name="point_state">
             <property name="text">
              <string>On / Off</string>
             </property>
            </widget>
           </item>
           <item>
            <layout class="QHBoxLayout" name="horizontalLayout_7">
             <item>
              <widget class="QLabel" name="point_intensity_label">
               <property name="text">
                <string>I</string>
               </property>
              </widget>
             </item>
             <item>
              <widget class="QSlider" name="point_intensity">
               <property name="layoutDirection">
                <enum>Qt::RightToLeft</enum>
               </property>
               <property name="orientation">
                <enum>Qt::Horizontal</enum>
               </property>
               <property name="tickPosition">
                <enum>QSlider::NoTicks</enum>
               </property>
              </widget>
             </item>
            </layout>
           </item>
           <item>
            <layout class="QHBoxLayout" name="horizontalLayout_4">
             <item>
              <widget class="QLabel" name="point_x">
               <property name="sizePolicy">
                <sizepolicy hsizetype="Fixed" vsizetype="Preferred">
                 <horstretch>0</horstretch>
                 <verstretch>0</verstretch>
                </sizepolicy>
               </property>
               <property name="text">
                <string>x</string>
               </property>
              </widget>
             </item>
             <item>
              <widget class="QLineEdit" name="point_x_value">
               <property name="sizePolicy">
                <sizepolicy hsizetype="Fixed" vsizetype="Fixed">
                 <horstretch>0</horstretch>
                 <verstretch>0</verstretch>
                </sizepolicy>
               </property>
               <property name="maximumSize">
                <size>
                 <width>50</width>
                 <height>16777215</height>
                </size>
               </property>
               <property name="text">
                <string>0</string>
               </property>
              </widget>
             </item>
             <item>
              <widget class="QLabel" name="point_y">
               <property name="sizePolicy">
                <sizepolicy hsizetype="Fixed" vsizetype="Preferred">
                 <horstretch>0</horstretch>
                 <verstretch>0</verstretch>
                </sizepolicy>
               </property>
               <property name="text">
                <string>y</string>
               </property>
              </widget>
             </item>
             <item>
              <widget class="QLineEdit" name="point_y_value">
               <property name="sizePolicy">
                <sizepolicy hsizetype="Fixed" vsizetype="Fixed">
                 <horstretch>0</horstretch>
                 <verstretch>0</verstretch>
                </sizepolicy>
               </property>
               <property name="maximumSize">
                <size>
                 <width>50</width>
                 <height>16777215</height>
                </size>
               </property>
               <property name="text">
                <string>0</string>
               </property>
              </widget>
             </item>
             <item>
              <widget class="QLabel" name="point_z">
               <property name="sizePolicy">
                <sizepolicy hsizetype="Fixed" vsizetype="Preferred">
                 <horstretch>0</horstretch>
                 <verstretch>0</verstretch>
                </sizepolicy>
               </property>
               <property name="text">
                <string>z</string>
               </property>
              </widget>
             </item>
             <item>
              <widget class="QLineEdit" name="point_z_value">
               <property name="sizePolicy">
                <sizepolicy hsizetype="Fixed" vsizetype="Fixed">
                 <horstretch>0</horstretch>
                 <verstretch>0</verstretch>
                </sizepolicy>
               </property>
               <property name="maximumSize">
                <size>
                 <width>50</width>
                 <height>16777215</height>
                </size>
               </property>
               <property name="text">
                <string>0</string>
               </property>
              </widget>
             </item>
            </layout>
           </item>
          </layout>
         </item>
         <item>
          <widget class="QPushButton" name="point_change_color_button">
           <property name="minimumSize">
            <size>
             <width>0</width>
             <height>28</height>
            </size>
           </property>
           <property name="text">
            <string>Change color</string>
           </property>
          </widget>
         </item>
         <item>
          <layout class="QVBoxLayout" name="verticalLayout_3">
           <item>
            <widget class="QLabel" name="projector_label">
             <property name="font">
              <font>
               <weight>75</weight>
               <bold>true</bold>
              </font>
             </property>
             <property name="text">
              <string>Projector Light</string>
             </property>
            </widget>
           </item>
           <item>
            <widget class="QCheckBox" name="projector_state">
             <property name="text">
              <string>On / Off</string>
             </property>
            </widget>
           </item>
           <item>
            <layout class="QHBoxLayout" name="horizontalLayout_8">
             <item>
              <widget class="QLabel" name="projectot_intensity_label">
               <property name="text">
                <string>I</string>
               </property>
              </widget>
             </item>
             <item>
              <widget class="QSlider" name="projectot_intensity">
               <property name="layoutDirection">
                <enum>Qt::RightToLeft</enum>
               </property>
               <property name="orientation">
                <enum>Qt::Horizontal</enum>
               </property>
              </widget>
             </item>
            </layout>
           </item>
           <item>
            <layout class="QHBoxLayout" name="horizontalLayout_3">
             <item>
              <widget class="QLabel" name="projector_cutOff">
               <property name="maximumSize">
                <size>
                 <width>150</width>
                 <height>16777215</height>
                </size>
               </property>
               <property name="text">
                <string>Cut Off Angle</string>
               </property>
              </widget>
             </item>
             <item>
              <widget class="QLineEdit" name="projector_cutOff_value">
               <property name="sizePolicy">
                <sizepolicy hsizetype="Expanding" vsizetype="Fixed">
                 <horstretch>0</horstretch>
                 <verstretch>0</verstretch>
                </sizepolicy>
               </property>
               <property name="maximumSize">
                <size>
                 <width>50</width>
                 <height>16777215</height>
                </size>
               </property>
               <property name="text">
                <string>12.5</string>
               </property>
              </widget>
             </item>
             <item>
              <widget class="QLabel" name="projector_outerCutOff">
               <property name="maximumSize">
                <size>
                 <width>150</width>
                 <height>16777215</height>
                </size>
               </property>
               <property name="text">
                <string>Outer Cut Off Angle</string>
               </property>
              </widget>
             </item>
             <item>
              <widget class="QLineEdit" name="projector_pos_y">
               <property name="maximumSize">
                <size>
                 <width>50</width>
                 <height>16777215</height>
                </size>
               </property>
               <property name="text">
                <string>17.5</string>
               </property>
              </widget>
             </item>
            </layout>
           </item>
          </layout>
         </item>
         <item>
          <widget class="QPushButton" name="projector_change_color_button">
           <property name="minimumSize">
            <size>
             <width>0</width>
             <height>28</height>
            </size>
           </property>
           <property name="text">
            <string>Change color</string>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QLabel" name="maze_label">
           <property name="font">
            <font>
             <weight>75</weight>
             <bold>true</bold>
            </font>
           </property>
           <property name="text">
            <string>Maze</string>
           </property>
          </widget>
         </item>
         <item>
          <layout class="QHBoxLayout" name="horizontalLayout_5">
           <item>
            <widget class="QLabel" name="time_label">
             <property name="text">
              <string>Time</string>
             </property>
            </widget>
           </item>
           <item>
            <widget class="QLCDNumber" name="mins_elapsed">
             <property name="font">
              <font>
               <family>MS Shell Dlg 2</family>
              </font>
             </property>
            </widget>
           </item>
           <item>
            <widget class="QLCDNumber" name="secs_elapsed">
             <property name="font">
              <font>
               <family>MS Shell Dlg 2</family>
              </font>
             </property>
            </widget>
           </item>
          </layout>
         </item>
         <item>
          <layout class="QHBoxLayout" name="horizontalLayout_9">
           <item>
            <widget class="QCheckBox" name="show_path">
             <property name="text">
              <string>Path</string>
             </property>
            </widget>
           </item>
           <item>
            <widget class="QCheckBox" name="god_mode">
             <property name="text">
              <string>God mode</string>
             </property>
            </widget>
           </item>
          </layout>
         </item>
         <item>
          <layout class="QHBoxLayout" name="horizontalLayout_11">
           <item>
            <layout class="QHBoxLayout" name="horizontalLayout_10">
             <item>
              <widget class="QLabel" name="maze_size_label">
               <property name="text">
                <string>Maze size</string>
               </property>
              </widget>
             </item>
             <item>
              <widget class="QSpinBox" name="maze_size">
               <property name="minimum">
                <number>2</number>
               </property>
               <property name="maximum">
                <number>50</number>
               </property>
               <property name="value">
                <number>5</number>
               </property>
              </widget>
             </item>
            </layout>
           </item>
           <item>
            <widget class="QPushButton" name="Restart">
             <property name="text">
              <string>Restart</string>
             </property>
            </widget>
           </item>
          </layout>
         </item>
        </layout>
       </widget>
      </widget>
      <widget class="QSplitter" name="splitter_2">
       <property name="orientation">
        <enum>Qt::Horizontal</enum>
       </property>
      </widget>
     </widget>
    </item>
   </layout>
  </widget>
  <widget class="QStatusBar" name="statusbar"/>
 </widget>
 <customwidgets>
  <customwidget>
   <class>CubeWidget</class>
   <extends>QWidget</extends>
   <header location="global">cubewidget.h</header>
   <container>1</container>
   <slots>
    <signal>fpsChanged(double)</signal>
    <signal>setShowPathFlag(bool)</signal>
    <signal>setGodModeFlag(bool)</signal>
    <signal>minsElapsed(int)</signal>
    <signal>secsElapsed(int)</signal>
    <slot>stripsUpd(int)</slot>
    <slot>directLightToggle(int)</slot>
    <slot>pointLightToggle(int)</slot>
    <slot>projectorLightToggle(int)</slot>
    <slot>sceneNumObjChange(int)</slot>
    <slot>sceneMaterialChange(QString)</slot>
    <slot>showPath(bool)</slot>
    <slot>godMode(bool)</slot>
    <slot>restartScene()</slot>
    <slot>resizeScene(int)</slot>
   </slots>
  </customwidget>
 </customwidgets>
 <resources/>
 <connections>
  <connection>
   <sender>direct_state</sender>
   <signal>stateChanged(int)</signal>
   <receiver>cubewidget</receiver>
   <slot>directLightToggle(int)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>969</x>
     <y>67</y>
    </hint>
    <hint type="destinationlabel">
     <x>641</x>
     <y>73</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>point_state</sender>
   <signal>stateChanged(int)</signal>
   <receiver>cubewidget</receiver>
   <slot>pointLightToggle(int)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>667</x>
     <y>172</y>
    </hint>
    <hint type="destinationlabel">
     <x>591</x>
     <y>190</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>projector_state</sender>
   <signal>stateChanged(int)</signal>
   <receiver>cubewidget</receiver>
   <slot>projectorLightToggle(int)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>668</x>
     <y>309</y>
    </hint>
    <hint type="destinationlabel">
     <x>600</x>
     <y>312</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>cubewidget</sender>
   <signal>fpsChanged(double)</signal>
   <receiver>fps</receiver>
   <slot>setNum(double)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>645</x>
     <y>43</y>
    </hint>
    <hint type="destinationlabel">
     <x>968</x>
     <y>25</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>show_path</sender>
   <signal>toggled(bool)</signal>
   <receiver>cubewidget</receiver>
   <slot>showPath(bool)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>661</x>
     <y>485</y>
    </hint>
    <hint type="destinationlabel">
     <x>612</x>
     <y>524</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>god_mode</sender>
   <signal>toggled(bool)</signal>
   <receiver>cubewidget</receiver>
   <slot>godMode(bool)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>821</x>
     <y>493</y>
    </hint>
    <hint type="destinationlabel">
     <x>438</x>
     <y>507</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>Restart</sender>
   <signal>clicked()</signal>
   <receiver>cubewidget</receiver>
   <slot>restartScene()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>833</x>
     <y>510</y>
    </hint>
    <hint type="destinationlabel">
     <x>221</x>
     <y>397</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>cubewidget</sender>
   <signal>setShowPathFlag(bool)</signal>
   <receiver>show_path</receiver>
   <slot>setChecked(bool)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>564</x>
     <y>266</y>
    </hint>
    <hint type="destinationlabel">
     <x>696</x>
     <y>492</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>cubewidget</sender>
   <signal>setGodModeFlag(bool)</signal>
   <receiver>god_mode</receiver>
   <slot>setChecked(bool)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>578</x>
     <y>342</y>
    </hint>
    <hint type="destinationlabel">
     <x>852</x>
     <y>483</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>maze_size</sender>
   <signal>valueChanged(int)</signal>
   <receiver>cubewidget</receiver>
   <slot>resizeScene(int)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>774</x>
     <y>526</y>
    </hint>
    <hint type="destinationlabel">
     <x>349</x>
     <y>456</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>cubewidget</sender>
   <signal>minsElapsed(int)</signal>
   <receiver>mins_elapsed</receiver>
   <slot>display(int)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>587</x>
     <y>222</y>
    </hint>
    <hint type="destinationlabel">
     <x>798</x>
     <y>446</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>cubewidget</sender>
   <signal>secsElapsed(int)</signal>
   <receiver>secs_elapsed</receiver>
   <slot>display(int)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>421</x>
     <y>235</y>
    </hint>
    <hint type="destinationlabel">
     <x>903</x>
     <y>451</y>
    </hint>
   </hints>
  </connection>
 </connections>
</ui>
